src/main/scala/Joint.scala

```diff
@@ -49,12 +115,14 @@ object Joint {
   * of the line from the joint's center in the local coordinate system.
   */
 case class Joint(normalVec: (Double, Double, Double), localOrigin: (Double, Double, Double),
-                 center: (Double, Double, Double), dipAngle: Double, dipDirection: Double, phi: Double,
-                 cohesion: Double, shape: Seq[((Double, Double, Double),Double)]) {
+                 center: (Double, Double, Double), phi: Double, cohesion: Double,
+                 shape: Seq[((Double, Double, Double),Double)]) {
   val (a, b, c) = normalVec
   val (centerX, centerY, centerZ) = center
   val d = Joint.findDistance(normalVec, localOrigin, center)
   val (localX, localY, localZ) = localOrigin
+  val dipAngle = Joint.dipAngle(normalVec)
```

Comment:
That's a really good point. I don't think it's a premature optimization, it makes sense to just avoid these calcs if we can. I'll go ahead and add that.